src/modules/cluster/cluster-list/cluster-list.js

```diff
@@ -0,0 +1,73 @@
+(function() {
+    "use strict";
+
+    var app = angular.module("TendrlModule");
+
+    app.controller("clusterController", clusterController);
+
+    /*@ngInject*/
+    function clusterController($scope, $state, $interval, config, $filter, utils) {
+        var vm = this,
+            key,
+            len,
+            temp = [],
+            clusterData,
+            cluster,
+            i;
+
+        vm.importCluster = importCluster;
+
+        init();
+
+        function init() {
+            _createClusterList();
+        }
+
+        /*Refreshing list after each 30 second interval*/
+        var timer = $interval(function () {
+            
+            utils.getObjectList("cluster")
+                .then(function(data) {
+                    config.clusterData = data;
+                });
+            
+            init();
+
+        }, 1000 * config.initData.refreshIntervalTime );
+
+        /*Cancelling interval when scope is destroy*/
+        $scope.$on('$destroy', function() {
+            $interval.cancel(timer);
+        });
+
+        function importCluster() {
+            $state.go("import-cluster");
+        }
+
+        function _createClusterList() {
+            clusterData = config.clusterData.clusters;
```

Comment:
you should check this condition before doing any operation - config.clusterData !== null 
because in index.js file line no 9  , we are setting up "null" value for it ..  hence in case if api call get fails than  config.clusterData won't have data still it will point to null.